app/js/components/names.jsx

```diff
@@ -1,265 +1,251 @@
-                                 />
-                    </div>
-                </div>
-
-                <div className="form-group ">
-                    <label className="control-label col-sm-2">Middle Name  </label>
-                    <div className="col-sm-5">
-                        <input  className="form-control"
-                                ref="middlename"
-                                name="middleName"
-                                type="text"
-                                value={this.state.middleName}
-                                onChange={this.handleChange}
-                                readOnly={this.state.display === 'view' ? 'readonly' : null}
-                                />
-                    </div>
-                </div>
-
-                <div className="form-group ">
-                    <label className="control-label col-sm-2"> Family Name* </label>
-                    <div className="col-sm-5">
-                        <input  className="form-control"
-                                ref="familyname"
-                                name="familyName"
-                                type="text"
-                                value={this.state.familyName}
-                                onChange={this.handleChange}
-                                readOnly={this.state.display === 'view' ? 'readonly' : null}
-                                required />
-                    </div>
-                </div>
-
-                <div className="form-group ">
-                    <label className="control-label col-sm-2"> Created By: </label>
-                    <div className="col-sm-5">
-                        <input  className="form-control"
-                                name="createdby"
-                                type="text"
-                                size="50"
-                                value= {this.state.creator +'  ' + this.state.dateCreated}
-                                disabled />
-                    </div>
-                </div>
-                {this.state.display == "edit" &&
-                <div className="form-group">
-                    <div className="col-sm-2">
-                        <button type="submit"
-                             name="update"
-                             onClick={this.handleSave}
-                             className="btn btn-success form-control">
-                             Save</button>
-                    </div>
-                    <div className="col-sm-2">
-                        <button type="button"
-                             name="cancel"
-                             onClick={this.handleCancel}
-                             className="btn btn-default form-control cancelBtn">
-                             Cancel</button>
-                    </div>
-               </div>
-            }
-
-            </form>
-            </div>
-        );
-    };
+    if(!this.state.familyName) {
```

Comment:
I think we could refactor this. I see you setting the state with same value it had previously. Again, even the error state could be set in the first setState call

Reply:
Well spotted, I have changed both